dist/utils.js

```diff
@@ -183,27 +199,27 @@ angular.module('ts.utils').directive('tsTooltip', function ($templateCache, $tim
         switch (direction) {
           case 'right':
           case 'left':
-            arrowBoxContainer.style.top = topCommon + $element[0].offsetHeight - $scope.tooltipMain[0].offsetHeight / 2 - ARROW_SIZE + 'px';
+            arrowBoxContainer.style.top = topCommon + element[0].offsetHeight - $scope.tooltipMain[0].offsetHeight / 2 - element.outerHeight() / 2 + 'px';
             break;
           case 'top':
           case 'bottom':
-            arrowBoxContainer.style.left = leftCommon + $element[0].offsetWidth / 2 - $scope.tooltipMain[0].offsetWidth / 2 + 'px';
+            arrowBoxContainer.style.left = leftCommon + element[0].offsetWidth / 2 - $scope.tooltipMain[0].offsetWidth / 2 + 'px';
             break;
         }
 
         //Sets the specific left or top values for each direction
         switch (direction) {
           case 'right':
-            arrowBoxContainer.style.left = leftCommon + $element[0].offsetWidth + ARROW_SIZE + 'px';
+            arrowBoxContainer.style.left = leftCommon + element[0].offsetWidth + ARROW_SIZE + parseInt($scope.tsTooltipMargin) + 'px';
```

Comment:
I suggest you wrap all the sum part, then concatenate with `'px'` at the end to avoid confusion (the first 4 things are ints, last one is string only)... Same above/below too please if possible!